src/queue/index.js

```diff
@@ -85,32 +85,40 @@ export class Queue {
      * Push a message into the queue
      * @param {string} msg - The string containing message to be pushed into RMQ
      */
-	push(msg) {
+	async push(msg) {
 		if (isNotDefined(msg)) {
 			log.error('Invalid message! Skipping.');
 			return;
 		}
 
+		// Serializing the JSON Object
+		// In this architecture, it is assumed that data is passed as JSON
+		// 		objects are passed from the producers to consumers
 		const serializedMessage = JSON.stringify(msg);
 
 		if (isNotDefined(this.channelPromise)) {
 			Error.undefinedValue('Queue.push:: undefined channel.');
 		}
 
-		this.channelPromise.then(channel => {
-			if (isNotDefined(channel)) {
+		try {
+			const channel = await this.channelPromise;
+
+			if (!channel) {
 				Error.undefinedValue(
 					'Queue.push:: Unable to get channel from promise.'
 				);
 			}
-			channel.assertQueue(QUEUE_NAME, {durable: true})
-				.then(ok => {
-					if (isNotDefined(ok)) {
+
+			const queueAssertion =
+				await channel.assertQueue(QUEUE_NAME, {durable: true});
+
+			if (!queueAssertion) {
```

Comment:
Seems that this is uncaught - I expect rather than checking if `queueAssertion` is defined, you want to add a `try... catch` block around the `const queueAssertion = await... `.

Reply:
We want to ensure that queueAssertion returned is a valid object, and hence we do a check. How to do that via try catch? We can wrap up:
```javascript
+                       const queueAssertion =
+				await channel.assertQueue(QUEUE_NAME, {durable: true});
+
+			if (!queueAssertion) {
+				Error.undefinedValue(
+					'Queue.push:: Could not assert queue.'
+				);
+			}
```
inside a try catch, if that's the aim?

Reply:
Well, `assertQueue` returns a promise - so when used with await, it'll throw an error if it can't get the queue for some reason - which is why it needs to be caught - https://www.squaremobius.net/amqp.node/channel_api.html#channel_assertQueue